Source/WebCore/platform/graphics/mac/controls/ColorWellMac.mm

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "ColorWellMac.h"
+
+#if PLATFORM(MAC) && ENABLE(INPUT_TYPE_COLOR)
+
+#import "ColorWellPart.h"
+#import "ControlFactoryMac.h"
+#import "LocalCurrentGraphicsContext.h"
+#import <wtf/BlockObjCExceptions.h>
+
+namespace WebCore {
+
+ColorWellMac::ColorWellMac(ColorWellPart& owningPart, ControlFactoryMac& controlFactory, NSButtonCell* buttonCell)
+    : ButtonControlMac(owningPart, controlFactory, buttonCell)
+{
+}
+
+void ColorWellMac::updateCellStates(const FloatRect& rect, const ControlStyle& style)
+{
+    [m_buttonCell setBezelStyle:NSBezelStyleTexturedSquare];
+    ButtonControlMac::updateCellStates(rect, style);
+}
+
+void ColorWellMac::draw(GraphicsContext& context, const FloatRect& rect, float deviceScaleFactor, const ControlStyle& style)
```

Comment:
Can we remove `paintColorWell` from `ThemeMac.mm`?

Reply:
The function `paintColorWell()` is removed from `ThemeMac.mm`